packages/manager/src/features/CloudPulse/Utils/utils.ts

```diff
@@ -161,3 +170,71 @@ export const getAllDashboards = (
     isLoading,
   };
 };
+
+/**
+ * @param port
+ * @returns object with isValid boolean and errorMsg string
+ * @description Validates a single port and returns the error message
+ */
+export const isValidPort = (
+  port: string
+): { errorMsg: string | undefined; isValid: boolean } => {
+  let errorMsg = PORTS_ERROR_MESSAGE;
+
+  if (port === '') {
+    return { errorMsg: undefined, isValid: true };
+  }
+
+  // Check for leading zeros
+  if (!port || port.startsWith('0')) {
+    errorMsg = PORTS_RANGE_ERROR_MESSAGE;
+    return { errorMsg, isValid: false };
+  }
+
+  const convertedPort = parseInt(port, 10);
+  if (!(1 <= convertedPort && convertedPort <= 65535)) {
+    errorMsg = PORTS_RANGE_ERROR_MESSAGE;
+    return { errorMsg, isValid: false };
+  }
+
+  return { errorMsg: undefined, isValid: true };
+};
+
+/**
+ * @param ports
+ * @returns object with isValid boolean and errorMsg string
+ * @description Validates a comma-separated list of ports and sets the error message
+ */
+export const arePortsValid = (
+  ports: string
+): { errorMsg: string | undefined; isValid: boolean } => {
```

Comment:
```suggestion
): { errorMsg: string | undefined; isValid: boolean } => {
```

define a interface for this return type & use it everywhere